File: src/main/java/fr/lewon/bot/manager/mappers/GameInfoMapper.java
package fr.lewon.bot.manager.mappers;

import java.util.List;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import fr.lewon.bot.manager.entities.GameInfosDTO;
import fr.lewon.bot.manager.modele.GameInfos;

@Mapper(
		componentModel = "spring",
		uses = { BotInfosMapper.class })
public interface GameInfoMapper {

	public List<GameInfosDTO> gameInfosToDto(List<GameInfos> gameInfos);

	@Mapping(source = "runnerInfos", target = "botInfos")
	public GameInfosDTO gameInfosToDto(GameInfos gameInfos);

}

File: src/main/java/fr/lewon/bot/manager/mappers/GameInfoMapper.kt
package fr.lewon.bot.manager.mappers

import fr.lewon.bot.manager.entities.GameInfoDTO
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper(componentModel = "spring", uses = [BotInfosMapper::class])
interface GameInfoMapper {
    fun gameInfosToDto(gameInfos: MutableList<GameInfos?>?): MutableList<GameInfoDTO?>?
    @Mapping(source = "runnerInfos", target = "botInfos")
    fun gameInfosToDto(gameInfos: GameInfos?): GameInfoDTO?
}